libs/hiraya/src/components/Button/index.tsx

```diff
@@ -1,12 +1,24 @@
-import { Button } from '@mui/material';
+import { ButtonProps, ButtonStyled } from './style';
+import { buttonIconOnly } from './theme';
 
-interface ButtonProps {
-  label: string;
-  variant?: 'text' | 'outlined' | 'contained';
-}
+const Button: React.FC<ButtonProps> = ({ label, iconOnly, ...props }) => {
+  const renderContent = () => {
+    if (!iconOnly) {
+      return label;
+    }
+    return buttonIconOnly();
+  };
```

Comment:
```suggestion
  const renderContent = () => iconOnly ? buttonIconOnly() : label
  
  // or better
  
  <ButtonStyled>{iconOnly ? buttonIconOnly() : label}</ButtonStyled>
```

I get what you're trying to do using if guard to catch default events, but in this case, this is doable with a ternary condition so it would be better if we approach this solution with a ternary instead and also define it inside the ButtonStyled itself like this